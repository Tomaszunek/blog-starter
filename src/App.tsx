import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import './App.css';


export namespace App {
  export interface IProps extends RouteComponentProps<void> {
    numbers: any   
  }
}

export default class App extends React.Component<App.IProps> {    
  public render() {
    return (
      <div className="App">
        abc
      </div>
    );
  }
}


