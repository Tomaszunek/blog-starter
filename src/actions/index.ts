export * from './articles';
export * from './products';
export * from './mainpage';
export * from './abilities';
export * from './projects';
export * from './appSystem';

