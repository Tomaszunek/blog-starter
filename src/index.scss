@import 'src/styles/variables.scss';

body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
  background-color: $body-background;
}

.container {
  padding: 0px 100px;
  background-color: $primary-background;

  @media only screen and (min-width: $sm) and (max-width: $lg) {
    padding: 0px 80px;
  }
  @media only screen and (min-width: $xs) and (max-width: $sm) {
    padding: 0px 60px;
  }
  @media only screen and (max-width: $xs) {
    padding: 0px 40px;ab
  }   
}