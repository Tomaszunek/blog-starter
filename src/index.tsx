import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Application } from './router';
import { configureStore } from './store';
import { createBrowserHistory } from 'history';
import registerServiceWorker from './registerServiceWorker';
import { Provider } from 'react-redux';
import { Router, Switch } from 'react-router';
import Navigation from 'src/pages/Navigation';
import Footer from 'src/pages/Footer';
import './index.scss';

const store = configureStore();
const history = createBrowserHistory();

ReactDOM.render(
  <Provider store={store}>    
    <Router history={history}>
      <div>
        <Navigation/>
          <Switch>
            <Application />
          </Switch>
        <Footer/>
      </div>
    </Router>   
  </Provider>,  
  document.getElementById('root') as HTMLElement
);
registerServiceWorker();