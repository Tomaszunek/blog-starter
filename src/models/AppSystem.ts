export interface IAppSystem {
    loading: boolean   
} 