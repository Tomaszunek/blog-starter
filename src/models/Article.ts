

export interface IArticle {
    id: number,
    name: string,
    body: string
}
    
  
export namespace ArticleFiltes {
    export enum Filter {
        SHOW_ALL = 'all',
        SHOW_ACTIVE = 'active',
        SHOW_COMPLETED = 'completed'
    }
}