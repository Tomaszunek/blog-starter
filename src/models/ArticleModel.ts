import { IContentCategory } from './ContentCategory';

export interface IArticleModel {
    id: number,
    name: string,
    slug: string,
    type: string,
    articleType: string,
    body: string,
    image: string,
    AppContentCategories?: Array<IContentCategory>,
    createdAt: Date,
    updatedAt: Date
}    
  
export namespace ArticleFiltes {
    export enum Filter {
        SHOW_ALL = 'all',
        SHOW_ACTIVE = 'active',
        SHOW_COMPLETED = 'completed'
    }
}