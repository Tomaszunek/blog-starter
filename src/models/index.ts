export * from './ArticleModel';
export * from './ProductModel';
export * from './ContentCategory';
export * from './MainPageContentModel';
export * from './AppSystem';

