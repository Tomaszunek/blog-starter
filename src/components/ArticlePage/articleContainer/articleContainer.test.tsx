import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import ArticleContainer from './articleContainer'
import { IArticleModel } from 'src/models';

describe('Error Page', () => {
  describe('Component should render right', () => {
    let app:any;    
    const articles:Array<IArticleModel> = [{
      id: 1,
      type: "ability",
      articleType: "ability",
      body: "body",
      createdAt: new Date(),
      image: "images/placeholder/abc.jpg",
      name: "woda",
      slug: "woda",
      updatedAt: new Date(),
    },{
      id: 2,
      type: "ability",
      articleType: "ability",
      body: "body",
      createdAt: new Date(),
      image: "images/placeholder/abc.jpg",
      name: "woda",
      slug: "woda",
      updatedAt: new Date(),
    }]
    beforeEach(() => {
      app = mount(<ArticleContainer  articles={articles} structure={[4,4]}/>);
    })
    it('render correctly snapshot component', () => {
      const tree = renderer
      .create(<ArticleContainer articles={articles} structure={[4,4]}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
    })
    it('expect two rendered tile', () => {
      expect(app.find('.articleTile').length).toEqual(2);
    })

    it('expect component elements', () => {
      expect(app.find('.articleConteiner').length).toEqual(1);
      expect(app.find('.grid.grid-four').length).toEqual(2);
    })
  });
});