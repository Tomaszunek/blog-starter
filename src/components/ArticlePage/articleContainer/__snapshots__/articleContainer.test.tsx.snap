// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page Component should render right render correctly snapshot component 1`] = `
<div
  className="articleConteiner"
>
  <div
    className="grid grid-four"
  >
    <div
      className="articleTile"
      data-attr={1}
    >
      <a
        href="/article/ability/woda"
      >
        <img
          alt=""
          src="./images/placeholders/images/placeholder/abc.jpg"
        />
        <div
          className="tileCont"
        >
          <div>
            <span>
              woda
            </span>
          </div>
        </div>
      </a>
    </div>
    <div
      className="articleTile"
      data-attr={2}
    >
      <a
        href="/article/ability/woda"
      >
        <img
          alt=""
          src="./images/placeholders/images/placeholder/abc.jpg"
        />
        <div
          className="tileCont"
        >
          <div>
            <span>
              woda
            </span>
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    className="grid grid-four"
  />
</div>
`;
